docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,312 @@
+
+## Specifying The React Element Type
+
+The first part of a JSX tag determines the type of the React element.
+
+Capitalized types indicate that the JSX tag is referring to a React component. These tags get compiled into a direct reference to the named variable, so if you use the JSX `<Foo />` expression, `Foo` must be in scope.
+
+Since JSX compiles into calls to `React.createElement`, the `React` library must also always be in scope from your JSX code.
+
+For example, both of the imports are necessary in this code, even though 'React' and 'MyComponent' are not directly referenced from JavaScript:
+
+```js
+import React from 'react';
+import MyComponent from './MyComponent';
+
+export default function MyRedComponent() {
+  return <MyComponent color="red" />;
+}
+```
+
+If you don't use a JavaScript bundler and added React as a script tag, it is already in scope as a React global.
+
+You can also refer to a React component using dot-notation from within JSX. This is convenient if you have a single module that exports many React components. For example, if `MyComponents.DatePicker` is a component, you can invoke it directly from JSX with:
+
+```js
+import React from 'react';
+
+var MyComponents = {
+  DatePicker: function(props) {
+    return <div>imagine a {props.color} datepicker here</div>;
+  }
+}
+
+export default function BlueDatePicker() {
+  return <MyComponents.DatePicker color={'blue'} />;
+}
+```
+
+If the element type starts with a lowercase letter, React will try to create a React element that maps directly to an HTML tag, rather than creating an element based on a custom component. In general, name your components starting with capital letters, and this will be fine. If you do have a component that starts with a lowercase letter, assign it to a capitalized variable before using it in JSX.
+
+For example, this code will not run as expected:
+
+```js
+import React from 'react';
+
+function hello(props) {
+  // This use of <div> is legitimate because div is a valid HTML tag
+  return <div>Hello {props.toWhat}</div>;
+}
+
+export default function HelloWorld() {
+  // This code attempts to create an HTML <hello> tag and fails
+  return <hello toWhat="World" />
+}
+```
+
+You cannot use a general expression as the React element type. If you do want to use a general expression to indicate the type of the element, just assign it to a capitalized variable first. For example, if you have a `findComponent()` function that returns a component, and you want to render one:
+
+```js
+import React from 'react';
+import findComponent from './findComponent';
+
+function render1() {
+  // Not valid JSX
+  return <findComponent() />;
+}
+
+function render2() {
+  var MyComponent = findComponent();
+
+  // Valid JSX
+  return <MyComponent />;
+}
+```
+
+## Props in JSX
```

Comment:
I'm conflicted about calling them props here.

If we are talking about JSX, "attributes" seems more correct. Attributes usually become props (for custom components). However `key` and `ref` are attributes obeying JSX rules that don't become props. 

I don't feel strongly about this though. @spicyj What do you think?